Instructor can display and broadcast message to queue

cameron-gagnon opened the pull request:
This feature adds the ability for instructors to create and save a message at the top of a course queue. Instructors can choose to broadcast the message to everyone on the queue as well as have it displayed on the top of the queue, or just post their message silently. This closes #86. 

**Note**: the gif recording software didn't record with the best resolution. There are some graphical artifacts in the gifs that aren't present in the actual build

- Ability to broadcast message to all students on the queue (using incognito on the right window, so I couldn't opt in for the notifications instead of alert popups)
![broadcast_message](https://cloud.githubusercontent.com/assets/8933413/25553342/6015126c-2c7e-11e7-9fd1-2638b89a3f0c.gif)

- Display message at the top of all queues
![screen shot 2017-04-29 at 1 45 38 am](https://cloud.githubusercontent.com/assets/8933413/25553292/952f3820-2c7d-11e7-88e6-90e8049f783f.png)

- Clears the message and does not display the message info to students when all instructors are offline
![clear_message](https://cloud.githubusercontent.com/assets/8933413/25553272/f27fd332-2c7c-11e7-9f59-86957f9d6de5.gif)

- Follows with the responsive design of the site
![responsive](https://cloud.githubusercontent.com/assets/8933413/25553348/a2067ec2-2c7e-11e7-8915-ea219ced7f5a.gif)


- Any instructor can edit the message
- Blank messages, when entered, aren't displayed

mterwill commented:
This looks awesome; thanks so much for implementing it! A couple of thoughts:

1. Do you think a long-form message would ever be useful? I could see pinning a common Q/A, etc. being a plausible use for this feature.
1. To that same end, I'm not entirely sure automatically clearing the message out on queue close is ideal. I could see setting a message "queue is closed, but I'll stay to clear it off," then going offline.
1. How do you feel about changing the instructor read-only view to appear similar to the student view with an ability to edit? Perhaps something like this: ![image](https://cloud.githubusercontent.com/assets/5882053/25567294/743bad42-2db8-11e7-82c3-cfa42efd9d7f.png)
It'd save a little space on the top of the page on mobile and more closely mirrors the student view.

Not all must haves to merge, just considerations I had.

mterwill commented on app/channels/queue_channel.rb:
```diff
@@ -118,6 +135,13 @@ def broadcast_request_change(action, request)
+          action: action,
+          message: message
+      })
+  end
```
Looks like this function's only called once. I'd be in favor of just pulling the logic up to `update_instructor_message` unless you had a specific reason for the helper method. I know it's not _super_ pretty, but these `QueueChannel.broadcast_to`s are all over the file anyway.

cameron-gagnon commented:
In response to the critiques:

1. Yes, the input is now a textarea with no text limit.
2. The message is no longer cleared when the queue goes offline and the message is still displayed to students.
3. I definitely agree on this point. Moved the edit and broadcast actions to be next to the instructor's header. Also changed the 'Save and Broadcast' button to be just 'Broadcast' so the text doesn't wrap on some mobile devices.

- Preview of editing and saving a message
![edit_to_save](https://cloud.githubusercontent.com/assets/8933413/25782856/a00fba08-3317-11e7-9378-9a8421df4c38.gif)

- When there's no message from the instructors, nothing is displayed to the students.
![no_message_unseen](https://cloud.githubusercontent.com/assets/8933413/25782868/bd7e2de0-3317-11e7-8e92-92ec9a9ffd43.gif)
